src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -38,14 +36,17 @@
 @ComponentScan(basePackageClasses = StudyService.class)
 public class StudyController {
 
-    @Autowired
-    private StudyService studyService;
+    private final StudyService studyService;
+
+    public StudyController(StudyService studyService) {
+        this.studyService = studyService;
+    }
 
     @GetMapping(value = "/studies")
     @ApiOperation(value = "Get all studies")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of studies")})
-    public ResponseEntity<List<StudyInfos>> getStudyList() {
-        List<StudyInfos> studies = studyService.getStudyList();
+    public ResponseEntity<Flux<StudyInfos>> getStudyList() {
```

Comment:
What does this do ? instead of ```ResponseEntity<Flux<StudyInfos>>``` I would have expected the Flux to be the outer type

Reply:
according to https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html#webflux-ann-responseentity

> WebFlux supports using a single value reactive type to produce the ResponseEntity asynchronously, and/or single and multi-value reactive types for the body.

So it should be Mono<ResponseEntity<Flux<StudyInfos>>

Reply:
Maybe we can remove (separate PR?) all the ResponseEntity to simplify things ? I don't think we need it because @ResponseBody should put the correct JSON content-type headers, and we can use exceptions for http statuses. Then we can only return Flux<StudyInfo> or Mono<Study> or other direct types

Reply:
> Maybe we can remove (separate PR?) all the ResponseEntity to simplify things ? I don't think we need it because @responsebody should put the correct JSON content-type headers, and we can use exceptions for http statuses. Then we can only return Flux or Mono or other direct types

ok i'll do that in a separate PR.